### ISSUE TITLE
[nexus 3.75+] Cannot create new repositories

### ISSUE THREAD

**Issue by FMotalleb** [sonatype-nexus-community/nexus-repository-composer#171]
Recently I've upgraded our Nexus instance to 3.75 and 3.75.1

>
>Now Nexus has composer proxy support (Pro only)
>

I think because the composer is included inside Nexus itself now, it's impossible to create new composer repositories.  
My older repositories are still available and working fine, so I don't have any real issues currently. But I'm afraid that this might change in the future, and with the next Nexus update, we might lose our composer (proxy and hosted) repositories, which my company relies on.

**Comment by stklcode** [comment]
I assume you're running a Pro instance, so both plugins are loaded?
Any error message or logs?

Just ran a quick test on a 3.75.1 OSS Container, I can create all 3 repo types there (hosted, proxy, group).

Quoting the documentation
> Nexus Repository native Composer support does not support migration from the Community plugin

OK ... Just quickly skimmed over the native class structures, the stuff looks (unsurprisingly) pretty similar, so I think there are conflicts with the recipes if both plugins are loaded.  Wondering which code is actually used for existing proxy repos :thinking:

Maybe it's enough to give our recipes unique names, like a `-community` suffix is enough, but if we have to rewrite all keys it could easily become a mess.

Removing `nexus-flags-feature-3.75.1-01-features.xml` feature could be a temporary solution, so the Pro plugin is not loaded.

**Comment by FMotalleb** [comment]
It should not be a pro instnce, Iam running `sonatype/nexus3:3.75.1-java17-ubi`

```
# While writing this, I wasn't sure if it was a Pro instance or not, so I checked the configuration. It appears to be the Pro edition, but I am confused because, during initialization, the logs indicate it is trying to load the OSS version + I don't have any license.

## DO NOT EDIT - CUSTOMIZATIONS BELONG IN $data-dir/etc/nexus.properties
##
# Jetty section
application-port=8081
application-host=0.0.0.0
nexus-args=${jetty.etc}/jetty.xml,${jetty.etc}/jetty-http.xml,${jetty.etc}/jetty-requestlog.xml
nexus-context-path=/${NEXUS_CONTEXT}

# Nexus section
nexus-edition=nexus-pro-edition
nexus-features=\
 nexus-pro-feature
 ```

**Comment by FMotalleb** [comment]
Looks like it worked (kinda)
![image](https://github.com/user-attachments/assets/752f3264-291c-41d0-bad8-2cc062485a48)

I'm still unable to create new composer repositories

**Comment by stklcode** [comment]
Any specific error message or logs?

**Comment by FMotalleb** [comment]
Container log
```
nexus-1  | 2024-12-20 13:04:58,025+0000 INFO  [FelixStartLevel] *SYSTEM org.sonatype.nexus.bootstrap.ConfigurationBuilder -   nexus.format.composer.enabled='false'
nexus-1  | 2024-12-20 13:05:52,996+0000 INFO  [FelixStartLevel] *SYSTEM org.sonatype.nexus.extender.NexusBundleTracker - ACTIVATING org.sonatype.nexus.plugins.nexus-repository-composer [0.1.5]
nexus-1  | 2024-12-20 13:05:53,089+0000 INFO  [FelixStartLevel] *SYSTEM org.sonatype.nexus.extender.NexusBundleTracker - ACTIVATED org.sonatype.nexus.plugins.nexus-repository-composer [0.1.5]
nexus-1  | 2024-12-20 13:06:05,205+0000 INFO  [FelixStartLevel] *SYSTEM org.sonatype.nexus.quartz.internal.task.QuartzTaskInfo - Task 'Cleanup unused composer blobs from nexus' [assetBlob.cleanup] : state=WAITING
nexus-1  | 2024-12-20 13:06:14,042+0000 INFO  [jetty-main-1] *SYSTEM org.sonatype.nexus.rapture.internal.RaptureWebResourceBundle -   nexus-repository-composer
nexus-1  | 2024-12-20 13:06:31,773+0000 ERROR [qtp420807980-109] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerHostedRecipe
nexus-1  | 2024-12-20 13:06:31,774+0000 ERROR [qtp420807980-109] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerProxyRecipe
nexus-1  | 2024-12-20 13:06:31,775+0000 ERROR [qtp420807980-109] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerGroupRecipe
nexus-1  | 2024-12-20 13:06:33,180+0000 ERROR [qtp420807980-108] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerHostedRecipe
nexus-1  | 2024-12-20 13:06:33,181+0000 ERROR [qtp420807980-108] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerProxyRecipe
nexus-1  | 2024-12-20 13:06:33,182+0000 ERROR [qtp420807980-108] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerGroupRecipe
nexus-1  | 2024-12-20 13:06:40,289+0000 ERROR [qtp420807980-702] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerHostedRecipe
nexus-1  | 2024-12-20 13:06:40,290+0000 ERROR [qtp420807980-702] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerProxyRecipe
nexus-1  | 2024-12-20 13:06:40,290+0000 ERROR [qtp420807980-702] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerGroupRecipe
nexus-1  | 2024-12-20 13:17:22,656+0000 ERROR [qtp420807980-700] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerHostedRecipe
nexus-1  | 2024-12-20 13:17:22,657+0000 ERROR [qtp420807980-700] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerProxyRecipe
nexus-1  | 2024-12-20 13:17:22,658+0000 ERROR [qtp420807980-700] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerGroupRecipe
nexus-1  | 2024-12-20 13:17:23,563+0000 ERROR [qtp420807980-703] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerHostedRecipe
nexus-1  | 2024-12-20 13:17:23,564+0000 ERROR [qtp420807980-703] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerProxyRecipe
nexus-1  | 2024-12-20 13:17:23,564+0000 ERROR [qtp420807980-703] admin org.sonatype.nexus.internal.datastore.DatabaseCheckImpl - Missing database version specified for class org.sonatype.nexus.repository.composer.internal.recipe.ComposerGroupRecipe
nexus-1  | 2024-12-20 13:30:00,019+0000 INFO  [quartz-9-thread-5] *SYSTEM org.sonatype.nexus.quartz.internal.task.QuartzTaskInfo - Task 'Cleanup unused composer blobs from nexus' [assetBlob.cleanup] state change WAITING -> RUNNING
nexus-1  | 2024-12-20 13:30:00,197+0000 INFO  [quartz-9-thread-5] *SYSTEM org.sonatype.nexus.quartz.internal.task.QuartzTaskInfo - Task 'Cleanup unused composer blobs from nexus' [assetBlob.cleanup] state change RUNNING -> WAITING (OK)
```

**Comment by stklcode** [comment]
This error has nothing to do with the Composer Pro feature.

Actually it’s fixed in #167, so using the current release 0.1.8 or later should resolve the issue.

**Comment by FMotalleb** [comment]
Ah, thanks, upgrading to v0.1.8 fixed the issue
